### PR TITLE
Pre migration work for NH, fixing an issue with the VT transform

### PR DESCRIPTION
**Pull request by JessicaWNava**
This PR implements the pre migration work for NH. 

Test Steps:
1. Review that the changes in organizations.yml match NH previous settings
2. Check that the resulting message matches the outbound message from the Covid pipeline
3. Make sure the tests pass

## Changes
- Added settings for NH to organizations.yml
- Created the NH receiver transform
- Added an integration test for NH
- Fixes issue with VT transform that was making all MSH changes in the transform not work.

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/settings/STLTs/NH/nh-dphs.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  featureFlags: null
+  keys: null
+  senders: []
+  receivers:
+    - name: "full-elr"
+      organizationName: "nh-dphs"
+      topic: "full-elr"
+      customerStatus: "active"
+      translation:
+        schemaName: "azure:/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml"
```
Comment (oslynn) [inline prime-router/settings/STLTs/NH/nh-dphs.yml:16]: Please remove "metadata".  It doesn't work in staging.  It should be "azure:/hl7_mapping ....

File: prime-router/settings/STLTs/NH/nh-dphs.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        - "Bundle.identifier.value.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.family.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.given.count() > 0"
+        - "Bundle.entry.resource.ofType(Patient).birthDate.exists()"
+        - "Bundle.entry.resource.ofType(Specimen).where(type.empty()).count() = 0"
+        - "Bundle.entry.resource.ofType(Patient).where(address.line[0].empty() and address.postalCode.empty() and telecom.where(system = 'phone').value.empty() and telecom.where(system = 'email').value.empty()).count() = 0"
+        - "((Bundle.entry.resource.ofType(Specimen).collection.collectedPeriod.exists() or Bundle.entry.resource.ofType(Specimen).collection.collected.exists())
+           or (Bundle.entry.resource.ofType(ServiceRequest).occurrence.exists() or Bundle.entry.resource.ofType(Observation).effective.exists()))"
+      routingFilter:
+              - "DoesNotMatch(test_performed_code, 85478-6,76080-1,92141-1,80383-3,82170-2,85477-8,76078-5,77026-3,77028-9,77027-1,92142-9,80382-5,82166-0,82167-8,82169-4,82168-6,92809-3,92808-5,95423-0)"
```
Comment (oslynn) [inline prime-router/settings/STLTs/NH/nh-dphs.yml:73]: Please remove line 73.

File: prime-router/settings/STLTs/NH/nh-dphs.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          onlyOncePerDay: false
+      description: ""
+      transport:
+        type: SFTP
+        host: sftp
+        port: 22
+        filePath: ./upload
+        credentialName: DEFAULT-SFTP
+      externalName: null
+      enrichmentSchemaNames: []
```
Comment (oslynn) [inline prime-router/settings/STLTs/NH/nh-dphs.yml:99]: Add usePid14ForPatientEmail transform here.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    resource: 'Bundle.entry.resource.ofType(Patient).extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.coding'
+    condition: '%resource.code.exists()'
+    value: [ '%resource.system.getCodingSystemMapping()' ]
+    hl7Spec: [ '/PATIENT_RESULT/PATIENT/PID-10-3' ]
+
+  - name: nh-unwanted-fields
+    value: [ '""' ]
+    hl7Spec: [ '%{ORC}-23-1', '%{ORC}-14-1', '%{OBR}-17-1', '%{OBX}-18-1', '%{OBX}-18-2', '%{OBX}-18-3', '%{PID}-13-1']
+
+  - name: nh-sending-application-namespace-id
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml:23]: I think, we can remove MSH-3 and let it be default.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  - name: nh-sending-application-universal-id
+    value: [ '"2.16.840.1.114222.4.1.237821"' ]
+    hl7Spec: [ '%{MSH}-3-2' ]
+
+  - name: nh-sending-application-universal-id-type
+    value: [ '"ISO"' ]
+    hl7Spec: [ '%{MSH}-3-3' ]
+
+  - name: nh-sending-facility-namespace-id
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml:35]: Use default MSH-4 as well.

File: prime-router/settings/STLTs/NH/nh-dphs.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        truncateHDNamespaceIds: false
+        truncateHl7Fields: null
+        usePid14ForPatientEmail: true
+        convertTimestampToDateTime: null
+        cliaForOutOfStateTesting: null
+        cliaForSender: {}
+        phoneNumberFormatting: "STANDARD"
+        suppressNonNPI: false
+        processingModeCode: null
+        replaceDiiWithOid: true
```
Comment (oslynn) [inline prime-router/settings/STLTs/NH/nh-dphs.yml:45]: Need to  add the relpaceDiiWithOid in enrichmentSchemaName as well.

File: prime-router/settings/STLTs/NH/nh-dphs.yml (hunk 1 above)
Comment (GilmoreA6) [reply prime-router/settings/STLTs/NH/nh-dphs.yml:16]: - same comment as on other PR's either use classpath: or follow Ott's suggestion and add the schema to azure

Comment (JessicaWNava) [reply prime-router/settings/STLTs/NH/nh-dphs.yml:16]: @GilmoreA6 We add the schemas to Azure in the validate tickets

File: prime-router/settings/STLTs/NH/nh-dphs.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        convertDateTimesToReceiverLocalTime: false
+        useHighPrecisionHeaderDateTimeFormat: false
+        type: "HL7"
+        truncationConfig:
+          truncateHDNamespaceIds: false
+          truncateHl7Fields: []
+          customLengthHl7Fields: {}
+      jurisdictionalFilter:
+        - "(Bundle.entry.resource.ofType(ServiceRequest)[0].requester.resolve().organization.resolve().address.state = 'NH') or (Bundle.entry.resource.ofType(Patient).address.state = 'NH')"
+        - "matches(processing_mode_code, P)"
```
[the remark this answers is not in the document]
Comment (JessicaWNava) [reply prime-router/settings/STLTs/NH/nh-dphs.yml]: Good catch, fixed.

**Review by oslynn — APPROVED** [review]

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+  - name: nh-message-profile-identifier-universal-id
+    value: [ '"2.16.840.1.113883.9.11"' ]
+    hl7Spec: [ '%{MSH}-21-3' ]
+
+  - name: nh-message-profile-identifier-universal-id-type
+    value: [ '"ISO"' ]
+    hl7Spec: [ '%{MSH}-21-4' ]
+
+  - name: nh-relpaceDiiWithOid-18-3
```
Comment (GilmoreA6) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml:71]: you want to have the condition on this only be true if the equipment identifier is actually valued. As written, you'll get 
.....|||^^^2.16.840.1.113883.3.3719^ISO|| but the .3 and .4 are condition fields based on the .1 being present

**Review by victor-chaparro — CHANGES_REQUESTED** [review]

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+  - name: nh-message-profile-identifier-universal-id-type
+    value: [ '"ISO"' ]
+    hl7Spec: [ '%{MSH}-21-4' ]
+
+  - name: nh-observation
+    resource: Bundle.entry.resource.ofType(Observation)
+    condition: '%resource.count() > 0'
+    schema: classpath:/metadata/hl7_mapping/receivers/STLTs/NH/NH-observation-fields.yml
+    constants:
+      hl7SegmentGroup: 'PATIENT_RESULT/ORDER_OBSERVATION/OBSERVATION(%{resultIndex})'
```
Comment (victor-chaparro) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml:75]: Does this actually work? Does it suppress the fields for all OBX segments? I didn't think you could reference the `resultIndex` here. I thought you could only reference the `resultIndex` from inside the child schema `/NH-observation-fields.yml`

Comment (JessicaWNava) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-receiver-transform.yml]: It was not. I added fields to the FHIR that test this and made changes to fix it. Thank you

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-observation-fields.yml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+constants:
+  OBX: '/PATIENT_RESULT/ORDER_OBSERVATION/OBSERVATION(%{resultIndex})/OBX'
+
+elements:
+  - name: nh-unwanted-fields-obx
```
Comment (GilmoreA6) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NH/NH-observation-fields.yml:5]: Sorry I didn't catch this before, but the reliable way to do this is:

  - name: obx-equipment-instance-identifier
    condition: false

It just overrides the entire mapping to OBX-18

**Review by victor-chaparro — APPROVED** [review]

**Review by GilmoreA6 — APPROVED** [review]

**Pull request merged**